mesonbuild/build.py

```diff
@@ -1236,6 +1236,13 @@ def process_kwargs(self, kwargs):
             if self.gnu_symbol_visibility not in permitted:
                 raise InvalidArguments('GNU symbol visibility arg {} not one of: {}'.format(self.gnu_symbol_visibility, ', '.join(permitted)))
 
+        rust_dependency_map = kwargs.get('rust_dependency_map', {})
+        if not isinstance(rust_dependency_map, dict):
```

Comment:
Seems like yet another thing that @dcbaker would have to rebase his build.py refactor on. Perhaps we should try to get that in first?